File: cpp/4.cpp
//A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
//Find the largest palindrome made from the product of two 3-digit numbers.

#include <iostream>
#include <string>
#include <list>

using namespace std;

int main(){
    int i ,x ,z ,n ,cache ,biggest;
    string straightNumber;
    string reverseNumber;
    string straightNumberCache;
    std::list<int> numbersList;
    biggest = 0;

    for (i = 999 ;i > 100;i = i - 1){
        for (x = 998 ;x > 99;x = x - 1){
            cache = i * x;
            straightNumber = to_string(cache);
            //I did the straightNumberCache beacuse swap function overwrites the existing variable
            straightNumberCache = straightNumber;
            n = straightNumberCache.length();
            for (z = 0 ;z < n / 2 ;z = z + 1){
                swap(straightNumberCache[z], straightNumberCache[n - z - 1]);
            }
            if (straightNumber == straightNumberCache){
                numbersList.push_front(cache);
            }
        }
    }
    //In this for loop it finds the biggest number in the list
    for (int x : numbersList){
        if (x > biggest){
            biggest = x;
        }
    }
    cout << biggest;
    //The result is 906609
}

File: Python/4.py
# Short Question Description: !!!
# Author: @SelimGunal
# Finished on !!!
#A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
#Find the largest palindrome made from the product of two 3-digit numbers.

#Our reverseable numbers will be stored in here
listOfNumbers = []

for i in range (999 ,99 ,-1):
    for x in range (i ,99 ,-1):
        #We changed it to the string because the reverse for strings is much more easier
        straightNumber = str(i * x)
        reverseNumber = straightNumber[::-1]
        #If they are the same it will add numbers to our list
        if (straightNumber == reverseNumber):
            #Here we turn back to integer
            listOfNumbers.append(int(straightNumber))
#In here i reversed the list so it will be bigger to smaller
listOfNumbers.sort(reverse=True)
print(listOfNumbers[0])
#The result is 906609